lib/src/patch.dart

```diff
@@ -22,8 +23,9 @@ class Patch {
 
   /// Applies a series of non-overlapping patches to the text of a file.
   static String applyAll(SourceFile file, List<Patch> patches) {
-    var sortedPatches = patches.toList()
-      ..sort((a, b) => a.selection.compareTo(b.selection));
+    var sortedPatches = patches.toList();
+    mergeSort(sortedPatches,
+        compare: (a, b) => a.selection.compareTo(b.selection));
```

Comment:
I don't think you need to pass this, I think calling `compareTo()` is the default behavior.